### PR TITLE
[consumer] Mark module as stable

### LINKED ISSUE #9046: Stabilize module consumer
Before stabilizing a module, an approver or maintainer must make sure that the following criteria have been met for at least two successive minor version releases:

- [x] No open issues or PRs in the module that would require breaking changes
- [x] No TODOs in the module code that would require breaking changes
- [x] No deprecated symbols in the module
- [x] No symbols marked as experimental in the module
- [x] The module follows the [Coding guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md)

Please also make sure to publicly announce our intent to stabilize the module on:

- [x] The #otel-collector CNCF Slack Channel
- [x] The #opentelemetry CNCF Slack channel
- [x] A Collector SIG meeting (if unable to attend, just add to the agenda)

To help other people verify the above criteria, please link to the announcement and other links used to complete the above in a comment on this issue.

Once all criteria are met, close this issue by moving this module to the `stable` module set.

### PR DESCRIPTION
**Pull request by mx-psi**
#### Description

Mark `consumer` module as stable. This is now possible after #11491, since `consumererror` is a separate module now.

#### Link to tracking issue
Fixes #9046

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @open-telemetry/collector-approvers

**Review by songy23 — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Comment by bogdandrutu** [comment]
I would merge this after the release to not rush it. Are we ok with that?

**Comment by mx-psi** [comment]
> I would merge this after the release to not rush it. Are we ok with that?

Definitely, and regardless of the release let's at least wait for a few days at least in case somebody has an objection

**Review by VihasMakwana — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Comment by jade-guiton-dd** [comment]
It seems that the current API only permits passing signal-agnostic `Option`s to `NewLogs`/`NewMetrics`/`NewTraces`. How sure are we that we will never need or want to implement consumer options that are specific to one particular signal? (As a hypothetical and possibly impractical example, what if once tracing is integrated to the consumer module (#8804), we want to specify attributes from consumed metrics to be included as part of the trace?)

**Comment by mx-psi** [comment]
Fromt he 2024-10-28 Collector stability meeting, @bogdandrutu will take a look and see if we should do https://github.com/open-telemetry/opentelemetry-collector/pull/11492#issuecomment-2437585935

**Review by codeboten — APPROVED** [review]
Will merge after today's release if no one has objections

**Comment by mx-psi** [comment]
About https://github.com/open-telemetry/opentelemetry-collector/pull/11492#issuecomment-2437585935 publicly: we discussed @jade-guiton-dd's comment offline. 

The summary is: we can still do this by detecting whether an option is signal specific at construction time and returning an error. Jade's suggestion would still improve the API in that it would move a construction time error into a compile time error, but after trying this out it would be quite complex to implement, so we think we can leave the API as is.

TL;DR: **we are good to go** :)
